### PR TITLE
First version of tile overlay (AIC-287)

### PR DESCRIPTION
**Pull request by raigex**

### CODE REVIEW

File: map/src/main/kotlin/edu/artic/map/MapFragment.kt [hunk 1]
```diff
@@ -174,27 +175,13 @@ class MapFragment : BaseViewModelFragment<MapViewModel>() {
...
-                                        LatLng(41.880730, -87.621027)
-                                )
-                        ).zIndex(.11f)
-                        .image(BitmapDescriptorFactory.fromAsset("AIC_Floor1.png"))
-        )
+        tileOverlay = map.addTileOverlay(TileOverlayOptions()
+                .zIndex(0.2f)
+                .tileProvider(GlideMapTileProvider(requireContext(),1)))
+
+//        map.addTileOverlay(TileOverlayOptions().zIndex(0.3f)
```
Comment (Cliabhach) [inline map/src/main/kotlin/edu/artic/map/MapFragment.kt]: This is still needed for future work, as @raigex tells me.

File: map/src/main/kotlin/edu/artic/map/rendering/DebugTileProvider.kt [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import com.google.maps.android.geometry.Point
+import com.google.maps.android.projection.SphericalMercatorProjection
+import edu.artic.map.TextMarkerGenerator
+import edu.artic.map.ZOOM_MIN
+import edu.artic.map.museumBounds
+import timber.log.Timber
+import java.io.ByteArrayOutputStream
+import kotlin.math.pow
+
+class DebugTileProvider2(context: Context) : TileProvider {
```
Comment (Cliabhach) [inline map/src/main/kotlin/edu/artic/map/rendering/DebugTileProvider.kt]: This is a temporary file, I hope?

File: map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+import com.bumptech.glide.Glide
+import com.google.android.gms.maps.model.Tile
+import timber.log.Timber
+import java.io.IOException
+import kotlin.math.pow
+
+/**
+ * Description:
+ */
+class GlideMapTileProvider(private val context: Context,
```
Comment (Cliabhach) [inline map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt]: This is the `TileProvider` we'll be using, correct?

File: map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        return when {
+            x >= 0 && y >= 0 -> {
+                // calculate tile row count
+                val tileXCount = (2.0).pow(zoom)
+
+                // grab which tile based on row, what y number it is in the tile list, x position
+                val tileNumber = (y * tileXCount + x).toInt()
+                Timber.d("Tile X Count $tileXCount with number $tileNumber")
+
+                val path = "http://aic-mobile-tours.artic.edu/sites/default/files/floor-maps/tiles/floor$floor/zoom$zoom/tiles-$tileNumber.png"
```
Comment (Cliabhach) [inline map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt]: We'll get these urls from the `MapFloorDao` at some future point, iiuc. Please add a todo for switching to `https`.

File: map/src/main/kotlin/edu/artic/map/rendering/MapTileAssetProvider2.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import android.content.res.AssetManager
+import com.google.android.gms.maps.model.Tile
+import com.google.android.gms.maps.model.TileProvider
+import timber.log.Timber
+import java.io.IOException
+
+/**
+ * Description:
+ */
+class MapTileAssetProvider2(private val assetManager: AssetManager,
```
Comment (Cliabhach) [inline map/src/main/kotlin/edu/artic/map/rendering/MapTileAssetProvider2.kt]: No longer needed, correct?

File: map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt (hunk 3 above)
Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt]: This is the tile provider we will be using. it is also the tile provider that will be updated to use a dynamic url

File: map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt (hunk 4 above)
Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt]: will do so

Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt]: but see above comment

File: map/src/main/kotlin/edu/artic/map/MapFragment.kt (hunk 1 above)
Comment (raigex) [reply map/src/main/kotlin/edu/artic/map/MapFragment.kt]: correct

File: map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt (hunk 4 above)
Comment (Cliabhach) [reply map/src/main/kotlin/edu/artic/map/rendering/GlideMapTileProvider.kt]: fair enough.

**Pull request merged**